tnt/src/main/java/org/dcsa/conformance/standards/tnt/party/QueryParameterSpecificRule.java

```diff
@@ -0,0 +1,68 @@
+package org.dcsa.conformance.standards.tnt.party;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static org.dcsa.conformance.standards.tnt.checks.TntDataSets.VALID_DOCUMENT_TYPE_CODES;
+import static org.dcsa.conformance.standards.tnt.checks.TntDataSets.VALID_EQUIPMENT_EVENT_TYPES;
+import static org.dcsa.conformance.standards.tnt.checks.TntDataSets.VALID_SHIPMENT_EVENT_TYPES;
+
+public class QueryParameterSpecificRule implements QueryParamRule{
+
+  private final Map<String, Set<String>> queryParamMap = Map.of(
+      "SHIPMENT", Set.of("shipmentEventTypeCode", "documentTypeCode", "carrierBookingReference",
+      "transportDocumentID", "transportDocumentReference", "equipmentReference", "eventCreatedDateTime"),
+      "TRANSPORT", Set.of("transportDocumentReference", "transportEventTypeCode", "transportCallID", "vesselIMONumber",
+      "exportVoyageNumber", "carrierServiceCode", "UNLocationCode", "equipmentReference", "eventCreatedDateTime"),
+      "EQUIPMENT", Set.of("carrierBookingReference", "transportDocumentReference", "transportCallID",
+      "vesselIMONumber", "exportVoyageNumber", "carrierServiceCode", "UNLocationCode", "equipmentEventTypeCode", "equipmentReference", "eventCreatedDateTime")
+    );
```

Comment:
These 2 final class fields (queryParamMap & excludedQueryParams) can be made static, so they are not build up upon every initiation. They don't change, so it is fine. 
Can you also format this class?

Reply:
this is done